tools/serverpod_cli/test/database/extentions/migration_with_many_to_many_relation_pgsql_code_generation_test.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/database/create_definition.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:serverpod_service_client/serverpod_service_client.dart';
+import 'package:test/test.dart';
+
+import '../../generator/dart/client_code_generator/class_constructors_test.dart';
+
+void main() {
```

Comment:
as per @SandPod's suggestion: Having tables `A(id, name)` and `B(id, a -> A)`, then do *not* drop `A` when both `A` and `B.a` are dropped by the migration already (in which case there is nothing pointing into `A` anymore anyway).

Thus we need to take the "target tables" (new schema) into account for this check as well.

Reply:
Added a test and support for this cases.